core-wih/src/main/java/io/elimu/a2d2/corewih/OAuth2WorkItemHandler.java

```diff
@@ -0,0 +1,119 @@
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import io.elimu.a2d2.exception.WorkItemHandlerException;
+
+public class OAuth2WorkItemHandler implements WorkItemHandler {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OAuth2WorkItemHandler.class);
+	
+	@Override
+	public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+		String clientId = (String) workItem.getParameter("clientId");
+		String scope = (String) workItem.getParameter("scope");
+		String username = (String) workItem.getParameter("username");
+		String password = (String) workItem.getParameter("password");
+		String grantType = (String) workItem.getParameter("grantType");
+		String tokenUrl = (String) workItem.getParameter("tokenUrl");
+		List<String> missingValues = new ArrayList<>();
+		if (StringUtils.isEmpty(clientId)) {
+			missingValues.add("clientId");
+		}
+		if (StringUtils.isEmpty(username)) {
+			missingValues.add("username");
+		}
+		if (StringUtils.isEmpty(password)) {
+			missingValues.add("password");
+		}
+		if (StringUtils.isEmpty(scope)) {
+			scope = "offline_access";
+		}
+		if (StringUtils.isEmpty(grantType)) {
+			grantType = "password";
+		}
+		if (StringUtils.isEmpty(tokenUrl)) {
+			missingValues.add("tokenUrl");
+		}
+		if (!missingValues.isEmpty()) {
+			throw new WorkItemHandlerException("Missing mandatory parameters: " + missingValues);
+		}
+		String content = "grant_type=" + grantType + "&client_id=" + clientId 
+				+ "&scope=" + scope + "&username=" + username 
+				+ "&password=" + password;
+	    BufferedReader reader = null; 
+	    HttpsURLConnection connection = null;
+	    Map<String, Object> results = workItem.getResults();
+	    try {
+	        URL url = new URL(tokenUrl);
+	        connection = (HttpsURLConnection) url.openConnection();
+	        connection.setRequestMethod("POST");
+	        connection.setDoOutput(true);
+	        connection.setRequestProperty("Accept", "*/*");
+	        connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+	        connection.setRequestProperty("Authorization", 
+	        		"Basic " + Base64.getEncoder().encodeToString((clientId + ":").getBytes()));
+	        PrintStream os = new PrintStream(connection.getOutputStream());
+	        os.print(content);
+	        os.close();
+	        reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+	        String line = null;
+	        StringWriter out = new StringWriter(connection.getContentLength() > 0 ? connection.getContentLength() : 2048);
+	        while ((line = reader.readLine()) != null) {
+	            out.append(line);
+	        }
+	        String response = out.toString();
+	        Gson gson = new Gson();
+	        Map<?, ?> output = (Map<?, ?>) gson.fromJson(response, Map.class);
```

Comment:
The following fields are also in the result, but I'm not confident that we need them.
`token_type`, `session_state`, and `scope`

Reply:
Added them as outputs